### ISSUE TITLE
NoMethodError: method Stupidedi::Schema::CompositeElementUse.descriptor is abstract

### ISSUE THREAD

**Issue by austinmbrown** [irobayna/stupidedi#194]
I'm trying to generate an 837 ("005010X222A1") transaction and getting this NoMethodError:
```
NoMethodError: method Stupidedi::Schema::CompositeElementUse.descriptor is abstract
	from /usr/local/bundle/gems/stupidedi-1.4.0/lib/stupidedi/schema/abstract_use.rb:19:in `descriptor'
	from /usr/local/bundle/gems/stupidedi-1.4.0/lib/stupidedi/values/composite_element_val.rb:18:in `descriptor'
...
```

The line of my code that's throwing the error is the SV1 in the 2400 loop:
```
b.SV1(
          b.composite("HC", claim.service_code, claim.service_modifier, nil, nil, nil, claim.service_description),
          claim.amount, "UN", 1
        )
```

I'm not clear on what's wrong here. My usage of the composite element looks the same to me as in other places that don't throw an error, and the contents of the composite element don't seem to have an effect (ie `b.composite(nil)` does the same thing).

I'm more inclined to assume I'm doing something wrong than that I found a bug, but I'm at the point where I need some external expertise to tell. Thanks for any troubleshooting guidance you can offer.

**Comment by kputnam** [comment]
I think it's a bug in stupidedi where it's trying to report an error, so I guess it's both! I'm working on reproducing it and should have a fix shortly.

In the meantime, you could implement CompositeElementUse#descriptor to return the string "foo". That method is used to report where a validation error occurred, but you already know it's in SV1-01. 

Maybe the error message you'll get back will give enough information to figure out what's wrong. But otherwise, I should have a fix before long.

**Comment by kputnam** [comment]
I think the problem is SV1-01 is required to be present, and it's a composite element.

There's an annoying problem here when a composite element is missing but required, there's no location or position to report. Nnormally the position of the composite element is just reported as the position of the first component... but there are none because it's missing.

Until somewhat recently a lot of these messages didn't have a position reported, so it would just say "SV1-07 is required" and you'd have to know where to look. When you are using BuilderDsl you'll have a call stack so it's redundant. But when you're validating a file it's annoying because you wouldn't know which SV1 is the problem.

Maybe there's a way to use the parent's position (where SV1 begins, rather than where SV1-07 is supposed to be). But until then I'm going to just make something up so you'll at least get an error message instead of an internal error.

**Comment by austinmbrown** [comment]
But SV1-07 *is* present?

When I added CompositeElementUse#descriptor, I got:
```
IndexError: head of empty array
	from /usr/local/bundle/gems/stupidedi-1.4.0/lib/ruby/array.rb:19:in `head'
	from /usr/local/bundle/gems/stupidedi-1.4.0/lib/stupidedi/values/composite_element_val.rb:20:in `position'```

**Comment by austinmbrown** [comment]
Yeah, just saw that I read that wrong. Thanks for all your help!

**Comment by kputnam** [comment]
@austinmbrown I just pushed a fix to `master`. If you have time, could you check and let me know if your example at the top of the issue now reports "required element SV107 BLAH BLAH is missing"? If so, I'll close this issue.

**Comment by austinmbrown** [comment]
Success! 

```
Stupidedi::Exceptions::ParseError: required element SV107 COMPOSITE DIAGNOSIS CODE POINTER is blank at file
```

Thanks so much! 
🙌 🏅 🍕

**Comment by maxfurman** [comment]
We just ran into this bug today - seems like it's been fixed in master for nearly two years but not released. Any chance of a 1.4.1 release?

**Comment by kputnam** [comment]
Yeah, it's been a long time since the last release. Time really flies... I didn't think it was so long ago. I'll talk with @irobayna and see if we can get a release out soon. Thanks for pointing this out.

**Comment by kputnam** [comment]
@maxfurman We released 1.4.1 yesterday that includes this bug fix. Thanks for reminding us it wasn't released!
